### PR TITLE
Add `hex_metadata.config` licenses to output

### LINKED ISSUE #17: Also read license information from `hex_metadata.config`?
@voltone, continuing from https://github.com/voltone/rebar3_sbom/issues/16, but opening as a new issue since they're only slightly related, not the same thing.

This is more of a feature request.

As https://github.com/ninenines/cowboy/issues/1651 surfaced, it's possible non-`rebar3` libraries are being consumed by `rebar3_sbom` to generate the bill of materials. Taking this into account, and assuming Hex.pm as the _de facto_ online package manager for Erlang/OTP, would you be willing to allow extending the plugin to read the license information from `hex_metadata.config`, too?

(I'm trying to find more information on this file, or how it's referenced, to understand how to proceed, but opened this parallel discussion to try to find common ground between these two approaches - `.app.src` and `.app`/no-`rebar3`)

### PR DESCRIPTION
**Pull request by kivra-pauoli**
Closes #17.

## Motivation

Not all available license information for a given "package" is retrieved; only the one found in `.app.src`.

## Description

Hex.pm is the de facto package manager for BEAM packages.

As is, when packaging from `rebar3`, an `hex_metadata.config` file gets added to the release that is eventually downloaded
into `_builds`.

We take advantage of that, and the fact that Hex.pm is de facto, as mentioned, and extend the "fetch licenses" mechanism to include further license information.

### CODE REVIEW

**Comment by kivra-pauoli** [comment]
Lemme know if you'd prefer I added a simple test case that stimulated this part of the change (I see no changes, but I'd feel more comfortable knowing I didn't break anything - only real tests were done locally, with packages that return `undefined` licenses or have no `hex_metadata.config` available).

File: src/rebar3_sbom_prv.erl [hunk 1]
```diff
@@ -54,15 +54,34 @@ dep_info(Dep) ->
     Source = rebar_app_info:source(Dep),
     Details = rebar_app_info:app_details(Dep),
     Deps = rebar_app_info:deps(Dep),
+    Licenses0 = proplists:get_value(licenses, Details, []),
```
Comment (kivra-pauoli) [inline src/rebar3_sbom_prv.erl:57]: Mind the added `, []`, so we can `++` with the result from `hex_metadata_licenses/1`.

**Comment by kivra-pauoli** [comment]
@voltone, 👋.

**Comment by kivra-pauoli** [comment]
@voltone, 👋.

**Comment by voltone** [comment]
Sorry, I had to step away (for two months) to answer the door :)

Seriously, never enough time to maintain FOSS... :(

File: src/rebar3_sbom_prv.erl (hunk 1 above)
Comment (voltone) [reply src/rebar3_sbom_prv.erl:57]: When constructing the SBoM, `[]` and `undefined` are both considered an empty value, so if no licenses are found the `licenses` attribute will still be omitted.

**Comment by voltone** [comment]
> Lemme know if you'd prefer I added a simple test case that stimulated this part of the change

Yeah, testing is conspicuously absent in this project. I'm working on a replacement for this plugin and its Mix sibling, in a unified project. I want to support newer spec versions, JSON output, have test coverage, etc. One day...

Thanks for your contribution!

**Pull request merged**